janus_core/log.py

```diff
@@ -5,13 +5,83 @@
 
 from janus_core.janus_types import LogLevel
 
-FORMAT = """
-- timestamp: %(asctime)s
-  level: %(levelname)s
-  message: %(message)s
-  trace: %(module)s
-  line: %(lineno)d
-""".strip()
+
+class YamlFormatter(logging.Formatter):  # numpydoc ignore=PR02
+    """
+    Custom formatter to convert multiline messages into yaml list.
+
+    Parameters
+    ----------
+    fmt : Optional[str]
+        A format string in the given style for the logged output as a whole. Default is
+        defined by `FORMAT`.
+    datefmt : Optional[str]
+        A format string in the given style for the date/time portion of the logged
+        output. Default is taken from logging.Formatter.formatTime().
+    style : Literal['%', '{', '$']
+        Determines how the format string will be merged with its data. Can be one of
+        '%', '{' or '$'. Default is '%'.
+    validate : bool
+        If True, incorrect or mismatched fmt and style will raise a ValueError. Default
+        is True.
+    defaults : Optional[dict[str, logging.Any]]
+         A dictionary with default values to use in custom fields. Default is None.
+    *args
+        Arguments to pass to logging.Formatter.__init__.
+    **kwargs
+        Keyword arguments to pass to logging.Formatter.__init__.
+
+    Methods
+    -------
+    format(record)
+        Format log message to convert new lines into a yaml list.
+    """
+
+    FORMAT = (
+        "\n- timestamp: %(asctime)s\n"
+        "  level: %(levelname)s\n"
+        "  message: %(message)s\n"
+        "  trace: %(module)s\n"
+        "  line: %(lineno)d\n"
+    )
```

Comment:
Could just be
```suggestion
    FORMAT = ("""
- timestamp: %(asctime)s
  level: %(levelname)s
  message: %(message)s
  trace: %(module)s
  line: %(lineno)d
""")
```

Reply:
Ah wasn't sure if the (lack of) indentation was an issue, but seems fine, thanks